lib/glueby/contract/rpc.rb

```diff
@@ -10,6 +10,14 @@ def client
       def configure(config)
         @rpc = Tapyrus::RPC::TapyrusCoreClient.new(config)
       end
+
+      def switch_wallet(wallet)
+        before = client.config[:wallet]
+        client.config[:wallet] = wallet
+        result = yield(client)
+        client.config[:wallet] = before
+        result
```

Comment:
How about this way.

* If `switch_wallet()` gets block argument, it reverts the wallet after the block executed.
* If `switch_wallet()` never get block argument, it just switches the wallet.

Reply:
Done at 8baaafebcba30853226fb803fa2470778b8937e3

Reply:
> Should add 'rescue' or 'ensure' statement to recover wallet even if generating errors in yield.

How about this comment?

Reply:
> I think this method name switch_wallet is not suitable because this method switch wallet but revert it finally.

これは、「`switch_wallet`というメソッド名が適切ではない。メソッドの中でwalletを切り替えているが、最後は戻しているので結果としてswitchしていない」という意味で書きました。

switchするためのメソッドを実装する必要があるということではなく、単にメソッド名を変えたほうがよいという意図でした。例えば、`RPC#perform_as(wallet_id)`とか。

あるいは、`switch_wallet`したときにwalletを元に戻していますが、元に戻さないような実装にしても良いのではないかと考えています。

Reply:
> How about this comment?

I missed that. I will update it.

Reply:
> これは、「switch_walletというメソッド名が適切ではない。メソッドの中でwalletを切り替えているが、最後は戻しているので結果としてswitchしていない」という意味で書きました。

当初のコメントの意図は理解していると思います。

このメソッド名を考えたときの念頭には、`Kernel#open` があって、このメソッドはファイルをオープンしますが、ブロックを渡したときには、ブロック引数にファイルを受け取り、ブロックが終わるとクローズする挙動になります。

最初の実装では、ブロックを渡さないケースの利用がなかったので、実装しませんでした。そこで最初のコメントを頂き、ブロックを渡さないケースを実装しなかったため、メソッド名と挙動がミスマッチになったと思われたのではと理解し、ブロックを渡さないケースを実装する対応を提案しました。

しかし、実際には最初実装しなかったようにブロックを渡さないケースは要らないので、ご提案の通り、`RPC#perform_as()` に変えたほうが良いと思いました。 :+1:

Reply:
Fixed the method name issue at 6daae954e1ca637d3999c3f7b0b88d0fca1ddb4d 
Fixed the error handling issue at 373e0502ba0b60bfc6cb580cb3be1ebc868772e3